apps/api-server/src/routes/api/project.js

```diff
@@ -188,6 +203,162 @@ router.route('/')
+            const validStatusIds = await getValidStatuses(req.projectId, resource.statuses.map(status => statusMap[status.id]));
+            await newResource.setStatuses(validStatusIds);
+          }
+        } catch (error) {
+          errors.push({ step: 'Create resources', error: error.message });
+        }
+      }
+
+      // Update widget IDs in widget config
+      const updateWidgetIds = (obj) => {
+        for (const key in obj) {
+          if (typeof obj[key] === 'object' && obj[key] !== null) {
+            updateWidgetIds(obj[key]);
+          } else {
+            if (key === 'projectId') {
+              obj[key] = req.projectId;
+            }
+            if (key === 'resourceId') {
+              obj[key] = resourceMap[obj[key]];
+            }
+            if (key.includes('tag') || key.includes('Tag')) {
+              if (obj[key]) {
+                let tagValue = typeof obj[key] === 'number' ? obj[key].toString() : obj[key];
+                if (typeof tagValue === 'string' && tagValue !== '') {
+                  tagValue = tagValue.split(',').map(id => tagMap[id] || id).join(',');
+                  obj[key] = tagValue;
+                }
+              }
+            }
+
+            if (key.includes('status') || key.includes('Status')) {
+              if (obj[key]) {
+                let statusValue = typeof obj[key] === 'number' ? obj[key].toString() : obj[key];
+                if (typeof statusValue === 'string' && statusValue !== '') {
+                  statusValue = statusValue.split(',').map(id => statusMap[id] || id).join(',');
+                  obj[key] = statusValue;
+                }
+              }
+            }
+            if (key === 'choiceguideWidgetId') {
+              obj[key] = widgetMap[obj[key]];
+            }
+          }
+        }
+      };
+
+      // Update widget function
+      async function updateWidget(widgetId, updatedData) {
+        try {
+          await db.Widget.update(updatedData, {
+            where: { id: widgetId }
+          });
+        } catch (error) {
+          errors.push({ step: 'Update widget', error: error.message });
+        }
+      }
+
+      // Update widget IDs in new widgets
+      for (const widget of newWidgets) {
+        try {
+          const updatedData = { config: widget.config };
+          updateWidgetIds(updatedData);
+          await updateWidget(widget.id, updatedData);
+        } catch (error) {
+          errors.push({ step: 'Update widget IDs in new widgets', error: error.message });
+        }
+      }
+
+      // Revert the config resource settings
+      try {
+        const project = await db.Project.findOne({ where: { id: req.projectId } });
+        const newConfig = project?.config || {};
+        newConfig.resources = req.resourceSettings || {};
+
+        await project.update({ config: newConfig });
+      } catch (error) {
+        errors.push({ step: 'Revert config resource settings', error: error.message });
+      }
```

Comment:
Dit blok is nu erg onoverzichtelijk, zou je dit kunnen refactoren naar losse functies?